shared/editor/embeds/index.tsx

```diff
@@ -53,13 +53,31 @@ export type EmbedProps = {
     href: string;
     matches: RegExpMatchArray;
   };
-  integrations?: any;
+  embedIntegrations?: any[];
 };
 
 function matcher(Component: React.ComponentType<EmbedProps>) {
-  return (url: string): boolean | [] | RegExpMatchArray => {
+  return (url: string, embedConfig?: any): boolean | [] | RegExpMatchArray => {
     // @ts-expect-error not aware of static
-    const regexes = Component.ENABLED;
+    let regexes = Component.ENABLED;
+
+    if (embedConfig) {
+      const hostname = embedConfig?.settings.hostname;
+
+      try {
+        const url = new URL(hostname);
+        regexes = [
+          new RegExp(
+            `^${url.protocol}//${url.host.replace(
+              /\./g,
+              "\\."
+            )}/(?!proxy).*(title=\\\\w+)?`
```

Comment:
I like the idea of including this logic further down the stack as you have here, but I think this line is the giveaway that really it should remain in the `embeds/Diagrams.tsx` component – each embed will have it's own rules beyond the hostname that don't make sense to include here

Reply:
Yes! It was just an attempt to make it working. I just pushed the working version so I can improve it later, incorporating feedback. A lot of it is kind of _hardcoded_ right now. But thanks for this note since I had completely forgotten about it after initial revision :laughing: .Still looking out for better patterns.

Reply:
Fixed!